File: src/main/java/sawyern/cookiebot/commands/BotCommand.java
package sawyern.cookiebot.commands;

import discord4j.core.DiscordClient;
import discord4j.core.event.domain.message.MessageCreateEvent;
import sawyern.cookiebot.exception.CookieException;

import java.util.List;

public interface BotCommand {
    void execute(MessageCreateEvent event, List<String> args) throws CookieException;
    void subscribeCommand(DiscordClient client);
}

File: src/main/kotlin/sawyern/cookiebot/commands/BotCommand.kt
package sawyern.cookiebot.commands

import discord4j.core.GatewayDiscordClient
import discord4j.core.event.domain.message.MessageCreateEvent

interface BotCommand {
    fun execute(event: MessageCreateEvent, args: List<String>)
    fun subscribe(client: GatewayDiscordClient)
}
